core/chains/evm/gas/models.go

```diff
@@ -329,7 +329,7 @@ func bumpDynamicFee(cfg Config, lggr logger.Logger, currentTipCap, currentBaseFe
 
 	if currentTipCap != nil {
 		if currentTipCap.Cmp(maxGasPrice) > 0 {
-			lggr.Errorf("invariant violation: ignoring current tip cap of %s that would exceed max gas price of %s", currentTipCap.String(), maxGasPrice.String())
+			lggr.Warnf("AssumptionViolation: ignoring current tip cap of %s that would exceed max gas price of %s", currentTipCap.String(), maxGasPrice.String())
```

Comment:
I wasn't expecting `AssumptionViolation` to be logged at such a low level. Is this meant for different cases than standard programmer error panics?

Reply:
Not sure, I just changed the text here to match. This _shouldn't_ happen because its calculated elsewhere but it can happen if you coded an estimator wrongly. Maybe it shouldn't be called AssumptionViolation?

Reply:
I don't know. AssumptionViolation is something new since https://app.shortcut.com/chainlinklabs/story/21618/introduce-more-granular-log-levels
It sounds to me more like a class of errors to be mapped to other levels :shrug:. If we need users to report it to us, then warn may be too low for them to notice.